src/data/list/basic.lean

```diff
@@ -828,9 +828,19 @@ by { cases l₂, { contradiction, }, { rw list.last'_append_cons, exact h } }
 theorem head_eq_head' [inhabited α] (l : list α) : head l = (head' l).iget :=
 by cases l; refl
 
-theorem mem_of_mem_head' {x : α} : ∀ {l : list α}, x ∈ l.head' → x ∈ l
+theorem surjective_head [inhabited α] : surjective (@head α _) := λ x, ⟨[x], rfl⟩
+
+theorem surjective_head' : surjective (@head' α) := option.forall.2 ⟨⟨[], rfl⟩, λ x, ⟨[x], rfl⟩⟩
+
+theorem surjective_tail : surjective (@tail α) :=
+by { rintro (_|⟨a, l⟩), exacts [⟨[], rfl⟩, ⟨a::a::l, rfl⟩] }
```

Comment:
Could also write this
```suggestion
theorem surjective_tail : surjective (@tail α)
| [] := ⟨[], rfl⟩
| (a :: l) := ⟨a :: a :: l, rfl⟩
```
which feels more in line with the style of many other list proofs